ReactAndroid/src/main/java/com/facebook/react/modules/share/ShareModule.java

```diff
@@ -0,0 +1,188 @@
+
+  private ReactApplicationContext mContext;
+
+  public ShareModule(ReactApplicationContext reactContext) {
+    super(reactContext);
+    mContext = reactContext;
+  }
+
+  @Override
+  public String getName() {
+    return "ShareModule";
+  }
+
+  interface OnClickItemListener {
+    void onClick(DialogInterface dialog, String packageName);
+  }
+
+  /* package */ class ShareDialogListener implements OnClickItemListener, OnDismissListener {
+
+    private final Promise mPromise;
+    private boolean mPromiseResolved = false;
+
+    public ShareDialogListener(Promise promise) {
+      mPromise = promise;
+    }
+
+    @Override
+    public void onClick(DialogInterface dialog, String packageName) {
+      if (!mPromiseResolved && getReactApplicationContext().hasActiveCatalystInstance()) {
+        WritableMap result = new WritableNativeMap();
+        result.putString("action", ACTION_SHARED);
+        result.putString("packageName", packageName);
+        mPromise.resolve(result);
+        mPromiseResolved = true;
+      }
+    }
+
+    @Override
+    public void onDismiss(DialogInterface dialog) {
+      if (!mPromiseResolved && getReactApplicationContext().hasActiveCatalystInstance()) {
+        WritableMap result = new WritableNativeMap();
+        result.putString("action", ACTION_DISMISSED);
+        mPromise.resolve(result);
+        mPromiseResolved = true;
+      }
+    }
+  }
+
+  /**
+   * Open a chooser dialog to send text content to other apps.
+   *
+   * Refer http://developer.android.com/intl/ko/training/sharing/send.html
+   * 
+   * @param content the data to send
+   * @param dialogTitle the title of the chooser dialog
+   */
+  @ReactMethod
+  public void share(ReadableMap content, String dialogTitle, Promise promise) {
+    if (content == null) {
+      promise.reject(ERROR_INVALID_CONTENT, "Content cannot be null");
+      return;
+    }
+
+    try {
+      Intent intent = new Intent(Intent.ACTION_SEND);
+      intent.setTypeAndNormalize("text/plain");
+      
+      if (content.hasKey("title")) {
+        intent.putExtra(Intent.EXTRA_SUBJECT, content.getString("title"));
+      }
+
+      if (content.hasKey("message")) {
+        intent.putExtra(Intent.EXTRA_TEXT, content.getString("message"));
+      } 
+
+      if (content.hasKey("url")) {
+        intent.putExtra(Intent.EXTRA_TEXT, content.getString("url")); // this will overwrite message
+      }
+
+      Activity activity = getCurrentActivity();
+      if (activity == null) {
+        promise.reject(
+            ERROR_NO_ACTIVITY,
+            "Tried to open a Share dialog while not attached to an Activity");
+        return;
+      }
+      
+      // We want to support both android.app.Activity and the pre-Honeycomb FragmentActivity
+      // (for apps that use it for legacy reasons). This unfortunately leads to some code duplication.
+      if (activity instanceof android.support.v4.app.FragmentActivity) {
```

Comment:
I don't think we need to support Honeycomb. React Native doesn't support anything prior to `4.1`